centml/compiler/backend.py

```diff
@@ -72,23 +66,22 @@ def _get_model_id(self, flow_graph: hidet.FlowGraph) -> str:
         return flow_graph_hash
 
     def _download_model(self, model_id: str) -> CompiledGraph:
-        download_response = requests.get(url=f"{server_url}/download/{model_id}", timeout=config_instance.TIMEOUT)
+        download_response = requests.get(
+            url=f"{config_instance.SERVER_URL}/download/{model_id}", timeout=config_instance.TIMEOUT
+        )
         if download_response.status_code != HTTPStatus.OK:
             raise Exception(
                 f"Download: request failed, exception from server:\n{download_response.json().get('detail')}"
             )
-
-        download_dir = os.path.join(base_path, model_id)
-        os.makedirs(download_dir, exist_ok=True)
-        download_path = os.path.join(download_dir, "cgraph.zip")
+        download_path = get_backend_compiled_forward_path(model_id)
+        os.makedirs(os.path.dirname(download_path), exist_ok=True)
```

Comment:
you can do this inside the utility function.

Reply:
we are already doing this inside the utility function right?

Reply:
I had it a bit differently but your way seems better